### PR TITLE
ACL Check in formData Controller

### PR DESCRIPTION
**Pull request by denduluriramasaimohanakhil**
lint: 0
   coverage :85%
ACLControllerCheck:
   lint :0
   coverage :85%

### CODE REVIEW

File: src/main/java/com/techsophy/tsf/runtime/form/controller/impl/FormDataControllerImpl.java [hunk 1]
```diff
@@ -1,29 +1,88 @@
...
+        return new ACLEvaluator(aclId,context,gatewayUrl);
+    }
+
+    private void checkACLRead(String formId, String relations)
+    {
+        String aclId=getAclId(formId).orElse(EMPTY_STRING);
+        if(StringUtils.hasText(aclId))
+        {
+            ACLEvaluator aclEvaluator=getAclEvaluatorObject(aclId,null,gatewayUrl);
+            if(aclEvaluator.getRead(tokenUtils.getTokenFromContext()).getDecision().equals(DENY))
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/controller/impl/FormDataControllerImpl.java]: what about undefined?

Comment (VaibhavTechsophy) [reply src/main/java/com/techsophy/tsf/runtime/form/controller/impl/FormDataControllerImpl.java]: in all below instances also, can we keep the code reusable

Comment (VaibhavTechsophy) [reply src/main/java/com/techsophy/tsf/runtime/form/controller/impl/FormDataControllerImpl.java]: can u write a function like checkACL(formId, relations, rules)
With rules being read/update/delete.

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

File: src/main/java/com/techsophy/tsf/runtime/form/controller/impl/FormDataControllerImpl.java (hunk 1 above)
Comment (denduluriramasaimohanakhil) [reply src/main/java/com/techsophy/tsf/runtime/form/controller/impl/FormDataControllerImpl.java]: Implemented as mentioned above

File: src/main/java/com/techsophy/tsf/runtime/form/controller/impl/FormDataControllerImpl.java [hunk 2]
```diff
@@ -70,26 +91,62 @@ else if ((StringUtils.hasText(sortBy) || StringUtils.hasText(sortOrder)) || (pag
...
+                        break;
+                    case UPDATE_RULE:
+                        decision = aclEvaluator.getUpdate(tokenUtils.getTokenFromContext());
+                        break;
+                    case DELETE_RULE:
+                        decision = aclEvaluator.getDelete(tokenUtils.getTokenFromContext());
+                        break;
+                    default: break;
+                }
+            assert decision != null;
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/controller/impl/FormDataControllerImpl.java:145]: good

**Review by VaibhavTechsophy — APPROVED** [review]

**Pull request merged**